### PR TITLE
[WIP] fix alugrid tests after codegen fix

### PR DESCRIPTION
**Pull request by renemilk**

### CODE REVIEW

**Comment by renemilk** [comment]
As feared a lot of the Alugrid tests were actually broken all this time, just not compiled/executed.
```
86% tests passed, 9 tests failed out of 66

Total Test time (real) = 1148.41 sec

The following tests FAILED:
	 35 - test_linearelliptic__cg_discretization (Failed)
	 36 - test_linearelliptic__swipdg_discretization (Failed)
	 39 - test_mpi_linearelliptic__swipdg_discretization (Failed)
	 40 - test_mpi_linearelliptic__swipdg_discretization-mpi-2 (Failed)
	 41 - test_mpi_linearelliptic__swipdg_estimators (Failed)
	 42 - test_mpi_linearelliptic__swipdg_estimators-mpi-2 (Failed)
	 43 - test_mpi_projections__lagrange_and_global (Failed)
	 51 - test_mpi_projections_part_1_a (Failed)
	 55 - test_mpi_projections_part_2_a (Failed)
Errors while running CTest
```
Starting with 35:
```
[  FAILED  ] 12 tests, listed below:
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__AO2013TestCase_AluConform2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__AO2013TestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__ER2007TestCase_AluConform2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__ER2007TestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__ESV2007TestCase_AluConform2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__ESV2007TestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__MixedBoundaryTestCase_AluSimplex2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__MixedBoundaryTestCase_AluSimplex2dGridType__gdt_istl_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__MixedBoundaryTestCase_AluConform2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__MixedBoundaryTestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__Spe10Model1TestCase_AluConform2dGridType__gdt_eigen_sparse.eoc_study
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__Spe10Model1TestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study

12 FAILED TESTS

```

Most of the failures seem harmless enough, just drifting expectations. However:
```
+==========================================================+
|+========================================================+|
||  Testcase: SPE10, Model1                               ||
||  (see http://www.spe.org/web/csp/datasets/set01.htm)   ||
|+--------------------------------------------------------+|
||  domain = [0, 5] x [0, 1]                              ||
||  diffusion: spe10 model 1 scalar data                  ||
||         |  2000 in [0.95, 1.10] x [0.30, 0.45]         ||
||  force: | -1000 in [3.00, 3.15] x [0.75, 0.90]         ||
||         | -1000 in [4.25, 4.40] x [0.25, 0.40]         ||
||  dirichlet = 0                                         ||
||  reference solution: discrete solution on finest grid  ||
|+========================================================+|
+==========================================================+

gdt.linearelliptic.discretization.cg.order_1
=======================================================================================
        grid         |    L2 (relative)    | H1_semi (relative)  |  energy (relative)  
---------------------+---------------------+---------------------+---------------------
     size |    width |    error |      EOC |    error |      EOC |    error |      EOC 
==========+==========+==========+==========+==========+==========+==========+==========
      160 | 3.54e-01 |      inf |     ---- |      inf |     ---- |      inf |     ----  (solve took 30ms)
----------+----------+----------+----------+----------+----------+----------+----------
      640 | 1.77e-01 |     -nan |     -nan |     -nan |     -nan |     -nan |     -nan  (solve took 104ms)
/home/r_milk01/master_gdt/dune-xt-common/dune/xt/common/test/common.cxx:101: Failure
Value of: actual_exponent
  Actual: -9223372036854775808
Expected: expected_exponent
Which is: -1
          Exponent comparison (in scientific notation, precision 2) failed for
          norm: L2
          actual_results[0]   = inf
          expected_results[0] = 0.77200000000000002
[  FAILED  ] linearelliptic_CG_discretization_Dune__GDT__LinearElliptic__Spe10Model1TestCase_AluConform2dGridType__gdt_istl_sparse.eoc_study (11824 ms)
```
Every SPE10 testcase is broken like that.

Also the inital 32 entitiy solution here seems way off:
```
+==========================================================+
|+========================================================+|
||  Testcase: mixed boundary types                        ||
|+--------------------------------------------------------+|
||  domain = [0, 1] x [0, 1]                              ||
||  diffusion = 1                                         ||
||  force     = 1                                         ||
||  neumann   = 0.1       on the right side               ||
||  dirichlet = 1/4 x y   everywhere else                 ||
||  reference solution: discrete solution on finest grid  ||
|+========================================================+|
+==========================================================+

gdt.linearelliptic.discretization.cg.order_1
=======================================================================================
        grid         |    L2 (relative)    | H1_semi (relative)  |  energy (relative)  
---------------------+---------------------+---------------------+---------------------
     size |    width |    error |      EOC |    error |      EOC |    error |      EOC 
==========+==========+==========+==========+==========+==========+==========+==========
       32 | 3.54e-01 | 1.12e+01 |     ---- | 1.27e+00 |     ---- | 1.27e+00 |     ----  (solve took 6ms)
----------+----------+----------+----------+----------+----------+----------+----------
      128 | 1.77e-01 | 6.60e-01 |     4.09 | 8.99e-01 |     0.50 | 8.99e-01 |     0.50  (solve took 23ms)
/home/r_milk01/master_gdt/dune-xt-common/dune/xt/common/test/common.cxx:101: Failure
Value of: actual_exponent
  Actual: 1
Expected: expected_exponent
Which is: -2
          Exponent comparison (in scientific notation, precision 2) failed for
          norm: L2
          actual_results[0]   = 11.199534355891346
          expected_results[0] = 0.038300000000000001

```
 Yay :sob:

**Comment by renemilk** [comment]
Actually, ER2007's also completely fubar:
```
+==================================================================+
|+================================================================+|
||  Testcase ESV2007: smooth data, homogeneous dirichlet          ||
||  (see testcase 1, page 23 in Ern, Stephansen, Vohralik, 2007)  ||
|+----------------------------------------------------------------+|
||  domain = [-1, 1] x [-1, 1]                                    ||
||  diffusion = 1                                                 ||
||  force     = 1/2 pi^2 cos(1/2 pi x) cos(1/2 pi y)              ||
||  dirichlet = 0                                                 ||
||  exact solution = cos(1/2 pi x) cos(1/2 pi y)                  ||
|+================================================================+|
+==================================================================+

gdt.linearelliptic.discretization.cg.order_1
=======================================================================================
        grid         |    L2 (relative)    | H1_semi (relative)  |  energy (relative)  
---------------------+---------------------+---------------------+---------------------
     size |    width |    error |      EOC |    error |      EOC |    error |      EOC 
==========+==========+==========+==========+==========+==========+==========+==========
       32 | 7.07e-01 | 1.57e-01 |     ---- | 3.77e-01 |     ---- | 3.77e-01 |     ----  (solve took 3ms)
----------+----------+----------+----------+----------+----------+----------+----------
      128 | 3.54e-01 | 1.00e+00 |    -2.67 | 1.00e+00 |    -1.41 | 1.00e+00 |    -1.41  (solve took 14ms)
/home/r_milk01/master_gdt/dune-xt-common/dune/xt/common/test/common.cxx:101: Failure
Value of: actual_exponent
  Actual: 0
Expected: expected_exponent
Which is: -2
          Exponent comparison (in scientific notation, precision 2) failed for
          norm: L2
          actual_results[1]   = 1
          expected_results[1] = 0.038199999999999998

```

File: dune/gdt/test/linearelliptic/eocexpectations/cg-er2007-2dalugrid.cxx [hunk 1]
```diff
@@ -30,15 +30,15 @@ std::vector<double> LinearEllipticEocExpectations<LinearElliptic::ER2007TestCase
 {
   if (type == "L2") {
 #if DXT_DISABLE_LARGE_TESTS
-    return {1.74e+00, 2.16e-01};
+    return {1.74e+00, 7.33e-01};
```
Comment (ftalbrecht) [inline dune/gdt/test/linearelliptic/eocexpectations/cg-er2007-2dalugrid.cxx]: A factor of 2 to 3 is quite a change. Do you have any idea happenned?

**Comment by renemilk** [comment]
Well, this is certainly getting weirder. Apparently nothing's actually broken on travis: https://travis-ci.org/dune-community/dune-gdt/builds/422525366

**Pull request merged**